render/render.js

```diff
@@ -101,13 +101,25 @@ module.exports = function($window) {
 		return element
 	}
 	function createComponent(parent, vnode, hooks, ns, nextSibling) {
-		vnode.state = Object.create(vnode.tag)
-		var view = vnode.tag.view
-		if (view.reentrantLock != null) return $emptyFragment
-		view.reentrantLock = true
-		initLifecycle(vnode.tag, vnode, hooks)
-		vnode.instance = Vnode.normalize(view.call(vnode.state, vnode))
-		view.reentrantLock = null
+		var sentinel
+		if (typeof vnode.tag === "function") {
+			vnode.state = null
+			sentinel = vnode.tag
+			if (sentinel.$$reentrantLock$$ != null) return $emptyFragment
+			sentinel.$$reentrantLock$$ = true
```

Comment:
@isiahmeadows ok. For reference, here's what always putting the lock on view looks like. I can rebase that and push it here if you want.

https://github.com/pygy/mithril.js/commit/7a30ca09943a02a05ada10c136d8fb00c3e3e278#diff-000bdfae56251b0715111d2b18c9de3c